### PR TITLE
#165413106 Create article

### PR DESCRIPTION
**Pull request by Niyitangasam**
#### Description

An Authenticated user can be able to create, update and delete an article and if  users create articles of the same title, the app should generate unique slugs such that the articles don't create conflict

#### Type of change
N/A
#### How Has This Been Tested?
This has been  tested by sending req specified in Readme 
```

- Create a new article:   POST  /api/v1/articles/
- Update existing article:  PUT   /api/v1/articles/{slug}
- Get article by slug:  GET  /api/v1/articles/{slug}
- Get  all articles:   GET   /api/v1/articles/
- Delete an article : DELETE  /api/v1/articles/{slug}

```

#### Checklist:
N/A
#### Pivotal Tracker ID
[#1165413106](https://www.pivotaltracker.com/n/projects/2326984/stories/165413106)

### CODE REVIEW

**Review by ericnyirimana — APPROVED** [review]
It is working well, good job

File: helpers/article.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+     *@return {boolean} Return true if user matche
+     */
+  static async foundArticleAndOwner(req) {
+    const currentSlug = req.params.slug;
+    const isExist = await Article.findOne({ where: { slug: currentSlug } });
+    if (!isExist) {
+      return 'Article Not found';
+    }
+    const authUser = await PassportHelper.findRecord(User, req.user.id);
+    const isOwner = ArticleHelper.checkId(authUser.dataValues.id, isExist.dataValues.UserId);
```
Comment (AJAkimana) [inline helpers/article.js]:
```suggestion
    const isOwner = authUser.dataValues.id === isExist.dataValues.UserId;
```
Then delete `checkIn` function

File: helpers/article.js [hunk 2]
```diff
@@ -72,22 +117,119 @@ class ArticleHelper {
...
+        {
+          title,
+          body,
+          description,
+          image,
+        },
+        {
+          where: { slug: req.params.slug }
+        }
+      );
```
Comment (AJAkimana) [inline helpers/article.js]: Try to write a few lines of the code in this file.

File: controllers/article.controller.js [hunk 3]
```diff
@@ -18,33 +15,10 @@ class ArticleController {
...
-      description,
-      image,
-      slug,
-      authorId: articleAuthor.id
-    });
+    const article = await ArticleHelper.createNewArticle(req);
     return res.status(201).send({
       status: res.statusCode,
-      data: { article, Author: userData }
+      data: { article: article.article, Author: article.userData }
```
Comment (macphilips) [inline controllers/article.controller.js]:
```js
{
     article: { ...article.article, author: article.userData }
}
```

**Pull request merged**